openpype/hosts/maya/plugins/publish/reset_xgen_attributes.py

```diff
@@ -0,0 +1,27 @@
+from maya import cmds
+
+import pyblish.api
+
+
+class ResetXgenAttributes(pyblish.api.InstancePlugin):
+    """Reset Xgen attributes."""
+
+    label = "Reset Xgen Attributes."
+    # Offset to run after workfile increment plugin.
+    order = pyblish.api.IntegratorOrder + 10.0
+    families = ["workfile"]
+
+    def process(self, instance):
+        for palette, data in instance.data.get("xgenAttributes", {}).items():
+            for attr, value in data.items():
+                node_attr = "{}.{}".format(palette, attr)
+                self.log.info(
+                    "Setting \"{}\" on \"{}\"".format(value, node_attr)
+                )
+                cmds.setAttr(node_attr, value, type="string")
+
+            cmds.setAttr(palette + "." + "xgExportAsDelta", True)
+
+        if instance.data.get("xgenAttributes", {}):
+            self.log.info("Saving changes.")
+            cmds.file(save=True)
```

Comment:
```suggestion
        xgen_attributes = instance.data.get("xgenAttributes", {})
        if not xgen_attributes :
            return

        for palette, data in xgen_attributes.items():
            for attr, value in data.items():
                node_attr = "{}.{}".format(palette, attr)
                self.log.info(
                    "Setting \"{}\" on \"{}\"".format(value, node_attr)
                )
                cmds.setAttr(node_attr, value, type="string")
            cmds.setAttr(palette + ".xgExportAsDelta", True)
            
        self.log.info("Saving changes.")
        cmds.file(save=True)
```

Opt out earlier.

Also, we're changing the scene with the `xgExportAsDelta` attribute for enabled here? are we sure we want to leave the user's workfile in a changed state after publishing? This should be avoided unless there's a hard requirement why this needs to be the case and will ALWAYS be what the user wants for their current workfile.

Reply:
Maya/Xgen changes this attribute when the incremental save triggers. We are changing it back to what it was before publishing. Found that the scene was not marked as modified either, so the user could end up saving the scene then on launching again, the attributes would be different.
Will comment on this though for prosperity.